server/controllers/class/addAnnouncement.ts

```diff
@@ -0,0 +1,29 @@
+import { NextFunction, Response } from 'express';
+import { addAnnouncementQuery } from '../../queries';
+import { CustomError, CustomRequest, addAnnouncementValidate } from '../../utils';
+
+const addAnnouncement = async (req: CustomRequest, res: Response, next: NextFunction) => {
+  try {
+    const { classId } = req.params;
+    const { description } = req.body;
+    const { role } = req.user;
+    if (role !== 'teacher') throw new CustomError(401, 'Unauthenticated');
```

Comment:
You already have a middleware to check if the user is a teacher or not, so just pass it before your controller and no need to this check